robot-command/src/main/java/org/obolibrary/robot/RenameCommand.java

```diff
@@ -151,17 +166,39 @@ public CommandState execute(CommandState state, String[] args) throws Exception
         CommandLineHelper.getBooleanValue(line, "allow-missing-entities", false);
 
     char separator;
+    Map<String, String> mappings = new HashMap<>();
+    Map<IRI, String> labels = new HashMap<>();
     // Process full renames
-    if (fullFile != null) {
-      separator = getSeparator(fullFile);
-      Map<String, String> mappings =
-          parseTableMappings(new File(fullFile), separator, allowDuplicates);
-      RenameOperation.renameFull(ontology, ioHelper, mappings, allowMissingEntities);
+    if (fullFile != null || singleMappings != null) {
+      // mappings from file
+      if (fullFile != null) {
+        separator = getSeparator(fullFile);
+        Map<String, List<String>> mappingsAndLabels =
+            parseTableMappings(new File(fullFile), separator, allowDuplicates);
+        for (Map.Entry<String, List<String>> map : mappingsAndLabels.entrySet()) {
+          String newIRI = map.getValue().get(0);
+          String newLabel = map.getValue().get(1);
+          mappings.put(map.getKey(), newIRI);
+          if (newLabel != null) {
+            labels.put(ioHelper.createIRI(newIRI), newLabel);
+          }
+        }
+      }
+      // mappings from individual args
+      if (singleMappings != null) {
```

Comment:
arent the singleMappings and fullFile parameters mutually exclusive? Maybe better have them in separate conditionals to avoid some unforeseen errors (I am thinking of filenames with spaces and other menaces).

Reply:
I didn't think they needed to be, if you had a file and wanted to add an extra mapping or two to it. @jamesaoverton what do you think?

Reply:
Now that I think of it, I actually have a use case for this case:

```
--mappings a b --mappings some.file.tsv --mappings c d
```

i.e. allowing the `--mappings` parameter to be properly n-ary with a mix of mapping files and hard coded mappings. This happens when you have a process that dynamically generates a mapping file for some specific set of terms, but you have a few hard coded mappings you want to change every time. Right now, I basically have multiple mapping tables, and then `cat` them together, which is fine. I just wanted to mention this in case you think others may be faced with the same kind of issue. (sorry this is the opposite of my first comment)

Reply:
The single mapping is `--mapping` while the file input is `--mappings` (plural). Is this too similar and confusing?

Reply:
ah no, no problem. Sorry didnt see that. No, if --mapping is n-ary (multiple or whatever that is called in CLI speech) its all good!